trace.go

```diff
@@ -138,9 +138,28 @@ func GetFields(err error) map[string]interface{} {
 	if err == nil {
 		return map[string]interface{}{}
 	}
+
+	// for proxyError combine top-level and nested fields.
+	if proxy, ok := err.(proxyError); ok {
+		fields := map[string]interface{}{}
+
+		// nested
+		for field, value := range GetFields(proxy.Err) {
+			fields[field] = value
+		}
+
+		// top-level
+		for field, value := range proxy.GetFields() {
+			fields[field] = value
```

Comment:
Is is possible to have overlapping field names? If so, which should take precedence?

Reply:
It is possible, though that shouldn't be common. It would require the invocation of `.AddFields()` method on an error returned from HTTP RPC call which already has some fields. At least in Teleport we rarely use fields, so I haven't seen such a pattern in code.

Given this scenario you'd expect the `.AddFields()` to override existing fields if there is any kind of conflict. This is exactly what the code above achieves.